src/llm/http_llm_calculator.cc

```diff
@@ -616,7 +617,10 @@ std::string HttpLLMCalculator::serializeUnaryResponse(const std::vector<std::str
     writer.String("choices");
     writer.StartArray();  // [
     int i = 0;
+    int n = this->request->getNumReturnSequences().value_or(1);
     for (const std::string& completeResponse : completeResponses) {
+        if (i >= n)
+            break;
```

Comment:
This is not enough. When applying `n` filter we must first sort `best_of` results and then pick `n` with highest score. 
See: https://github.com/mzegla/openvino.genai/blob/master/src/cpp/src/continuous_batching_pipeline.cpp#L259
I think we could do that after `read_all()` is called to create a subset of `n` best results.